src/_nebari/upgrade.py

```diff
@@ -314,13 +407,20 @@ def _version_specific_upgrade(
 
 
 class Upgrade_0_4_0(UpgradeStep):
+    """
+    Upgrade step for Nebari version 0.4.0
+
+    This class handles the specific upgrade tasks required for transitioning to version 0.4.0
+    """
+
     version = "0.4.0"
 
     def _version_specific_upgrade(
         self, config, start_version, config_filename: Path, *args, **kwargs
     ):
         """
-        Upgrade to Keycloak.
+        This version of Nebari introduces Keycloak for authentication, removes deprecated fields,
```

Comment:
These notes are good for all of the `_version_specific_upgrade_steps` but they should also be decorated with the `@override` decorator from `typing.override`. https://google.github.io/styleguide/pyguide.html#3831-overridden-methods

Reply:
Have added this decorator on all of the `_version_specific_upgrade_steps` in [`52ea69e`](https://github.com/nebari-dev/nebari/pull/2512/commits/52ea69e511a1ed309f03931edf857dd0914483dc)